src/docfx/lib/log/Telemetry.cs

```diff
@@ -120,6 +122,17 @@ public void Dispose()
             }
         }
 
+        private static (string fileExtension, string documentType, string mimeType) GetFileType(Document file)
+        {
+            var fileExtension = CoalesceEmpty(Path.GetExtension(file.FilePath.Path)?.ToLowerInvariant());
+            var mimeType = CoalesceEmpty(file.Mime.Value);
+            if (mimeType == "<null>" && file.ContentType == ContentType.Page && fileExtension == ".md")
+            {
+                mimeType = "Conceptual";
+            }
+            return (fileExtension, file.ContentType.ToString(), mimeType);
+        }
```

Comment:
need this be cached?

Reply:
good point, this function could be called plenty of times for the same file

---
In reply to: [402153515](https://github.com/dotnet/docfx/pull/5719#discussion_r402153515) [](ancestors = 402153515)